docs/requirements.txt

```diff
@@ -4,3 +4,4 @@ nbsphinx==0.6.1
 nbsphinx-link==1.3.0
 # needed for Pygments highlighting in notebooks
 ipython==7.13.0
+sphinx_rtd_theme==0.4.3
```

Comment:
I'm not entirely sue why our existing doc requirements were pinned?  rtd itself appears to install sphinx-rtd-theme<0.5

Reply:
Yeah, I don't quite know why either. I guess it ensures our docs don't break accidentally, but maybe it's suboptimal for this, because we won't automatically pick up improvements to the theme. However, I think these will overrule the RtDs defaults (that is, our requirements here will be the ones that are used for rendering the docs online).

For reference, it seems like RtD uses:

```
Pygments==2.3.1
setuptools==41.0.1
docutils==0.14
mock==1.0.1
pillow==5.4.1
alabaster>=0.7,<0.8,!=0.7.5
commonmark==0.8.1
recommonmark==0.5.0
sphinx<2
sphinx-rtd-theme<0.5
readthedocs-sphinx-ext<1.1
```

Notably, they're using a _very_ old Sphinx version there. Our CI is now installing `Sphinx==3.0.3`. Maybe we should add more things to our requirements to ensure CI and RtDs are more similar?

Reply:
I couldn't find any documentation on why rtd pre-installs specific old versions of packages, not sure that we want to add what they've already go in requirements.txt, as it may change?

Reply:
I've added the exact requirements specified by RtD into this PR, and opened https://github.com/stellargraph/stellargraph/pull/1537 to upgrade the versions to the current latest ones. I've done it this way so that this PR (theoretically) doesn't change anything at all about how our docs are rendered, and is just focused on getting accurate CI, and then #1537 can be independently tested and the rendering verified.